website/js/components/SourceReportsTile.js

```diff
@@ -0,0 +1,172 @@
+        let submissions = this.props.submissions;
+
+        // sort the submissions if this source specifies a sort function
+        if (this.props.reportBinding.sortBy) {
+            // (side note: we concat() to clone before sort()ing, because sort() mutates the array)
+            submissions = submissions.concat().sort(this.props.reportBinding.sortBy);
+        }
+
+        // create a per-submitter collapsible subsection within this source panel
+        const submitters = submissions.map((submissionData, idx) => {
+            // extract header fields, e.g. the submitter name
+            const submitterName = util.getFormattedFieldByProp(this.props.reportBinding.submitter.prop, submissionData);
+
+            // extract fields we care about from the submission data
+            const formattedCols = this.props.reportBinding.cols
+                .map(({ title, prop }) => ({
+                    title, prop, value: submissionData[prop]
+                }));
+
+            return (
+                <VariantSubmitter
+                    key={submissionData.id} idx={idx} submitter={submitterName} source={this.props.sourceName}
+                    reportBinding={this.props.reportBinding} cols={formattedCols} data={submissionData}
+                    hideEmptyItems={this.props.hideEmptyItems}
+                    onReportToggled={this.reportToggled}
+                    showHelp={this.showHelp}
+                    expanded={this.state.reportExpanded[idx]}
+                />
+            );
+        });
+
+        // create the source panel itself now
+        const groupTitle = `source-panel-${this.props.sourceName}`;
+        const header = (
+            <h3 style={{display: 'flex', flexDirection: 'row'}}>
+                <a style={{flexGrow: 1}} href="#" onClick={(event) => this.props.onChangeGroupVisibility(groupTitle, event)}>
+                    {this.props.groupTitle}
+                </a>
+
+                <a title='collapse all reports'
+                    onClick={(event) => this.setAllReportExpansion(event, false)}
+                    style={{cursor: 'pointer', marginRight: '10px'}}>
+                    <i className="fa fa-angle-double-up" aria-hidden="true" />
+                </a>
+
+                <a title='expand all reports'
+                    onClick={(event) => this.setAllReportExpansion(event, true)}
+                    style={{cursor: 'pointer'}}>
+                    <i className="fa fa-angle-double-down" aria-hidden="true" />
+                </a>
+            </h3>
+        );
+
+        // checks if we either have no submissions for this source, or if every field in every submission is empty
+        // (consideration: it's very unlikely that we're ever actually all empty, so should we skip this?)
+        // (use the cache if it's available, or else recalculate)
+        const allEmpty = this.allEmptyCached != null ? this.allEmptyCached : this.calculateAllEmpty(submissions);
```

Comment:
theres a lot to review here so it's hard to wrap my head around this completely -- is this just a check so that if there are no submissions or if all fields of a submission are empty we can refrain from rendering empty content on the page?

it's safe to say that for any variant with ClinVar or LOVD as a source, they will have submissions (reports) from the respective source. but, for example, if a variant has ClinVar as a source, it may not have submissions from LOVD and vice versa. it should also be safe to assume that if a variant has a submission, that submission will not be all empty fields.

please let me know if this doesn't address your concern!

Reply:
It does, thanks; if there's never a case in which a submission has all empty fields, then that check is unnecessary. I've removed it and will push an update in a minute or two.